optaplanner-core/src/test/java/org/optaplanner/core/api/score/holder/ScoreHolderTest.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.optaplanner.core.api.score.holder;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.optaplanner.core.api.score.Score;
+
+import static org.junit.Assert.assertNotNull;
+
+public class ScoreHolderTest {
+
+    @Rule
+    public final ExpectedException thrown = ExpectedException.none();
+
+    @Test
+    public void illegalStateExceptionThrownWhenConstraintMatchNotEnabled() {
+        ScoreHolder scoreHolder = buildScoreHolder(false);
```

Comment:
Yes, please refactor rename those to "expectedException". It's makes it easier to see that it's an alternative to the annotation parameter. It's also easier to deduce the type (ExceptedException). I think the JUnit examples were written in a jiffy.

It's a pity that ExpectedException ends with Exception but doesn't extend Exception... that's what really confused me. But "thrown" also confused me because I though it was what had been thrown, now what is expected to be thrown. Oh well, I guess "expectedExceptionRule" is confusing with Drools rules and "assertException" again hides the type... Let's stick with "expectedException"?

+1 for consistency in all usage of this approach. That's why I am think it's worth thinking twice on the name: it's a pattern for the future too.